### DISCUSSION TITLE
Renovatebot returning the wrong `mergeStateStatus`

### DISCUSSION THREAD

**Discussion by raspy8766** [renovatebot/renovate#11309] [Request Help]
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Describe the bug

Auto-merge is not working due to an incorrect merge status being reported by renovate bot. When I perform the [same query](https://github.com/renovatebot/renovate/blob/91db1b5c61deb059673a6d2ace84a458bec0da4e/lib/platform/github/index.ts#L550) that is used by the renovate bot, I get the following response stating the `mergeStateStatus` is `CLEAN`:
<details><summary>Open PR's GraphQL Query</summary>

```json
{
  "number": 2322,
  "headRefName": "renovate/testcafe-1.x",
  "baseRefName": "master",
  "title": "Update dependency testcafe to v1.15.2",
  "mergeable": "MERGEABLE",
  "mergeStateStatus": "CLEAN",
  "labels": {
      "nodes": []
  },
  "assignees": {
      "totalCount": 0
  },
  "reviewRequests": {
      "totalCount": 1
  },
  "commits": {
      "nodes": [
          {
              "commit": {
                  "author": {
                      "email": "bot@renovateapp.com"
                  },
                  "committer": {
                      "email": "bot@renovateapp.com"
                  },
                  "parents": {
                      "edges": [
                          {
                              "node": {
                                  "abbreviatedOid": "8655853",
                                  "oid": "8655853a93692b92ab91c1942f4aab2239df446f"
                              }
                          }
                      ]
                  }
              }
          }
      ]
  },
  "body": "...",
  "reviews": {
      "nodes": []
  }
},
```

</details>

There are no other authentication errors related errors, and it has been this way for about 3-4 months now. Any advice on how I might be able to fix/workaround this issue?

### Relevant debug logs

<details><summary>Logs</summary>

```bash
DEBUG: Processing existing PR(branch="renovate/testcafe-1.x")
DEBUG: Pull Request #2322 does not need updating(branch="renovate/testcafe-1.x")
DEBUG: PR is configured for automerge(branch="renovate/testcafe-1.x")
DEBUG: PR is not ready for merge(branch="renovate/testcafe-1.x")
{
  "canMergeReason": "mergeStateStatus = BLOCKED"
}
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
I think you have some wrong configuration, as renovate can successfully automerge on our renovate repos.

Please provide the logs and the config, or provide a minimal reproduction repo.

Please also format your description, there are no logs (only a markdown reference to a image as text) and the query isn't formatted. So please do not past logs as image, as they are not searchable

**Reply by raspy8766** [reply]
Thanks, my apologies for the bad image/text, I've updated it.

**Reply by viceice** [reply]
Check your branch protections, it's blocking renovate from automerge.

**Reply by raspy8766** [reply]
Okay, do you have to have them turned off completely; what change is expected there? I haven't seen anything in the Renovate docsite regarding that, did I skim over it somewhere?

**Reply by rarkins** [reply]
You should be reading GitHub documentation first, so that you understand how their branch protections work. Renovate has no rights to bypass branch protections so e.g. if you have required reviews or codeowners then GitHub will enforce that protection to all users and bots.

Also I think this title might be incorrect. Do you have any logs or proof that Renovate is returning the wrong `mergeStateStatus` or did you just assume that's the problem? `mergeStateStatus` can return that a branch is clean but that doesn't mean that branch protections have been met.

If the title is mistaken, please change it so that it doesn't mislead future readers. If you think it's not mistaken, please describe which log you're looking at.

**Reply by rarkins** [reply]
After you understand how GitHub's branch protections work, our documentation on it is here: https://docs.renovatebot.com/automerge-configuration/#frequent-problems-and-how-to-resolve-them

**Reply by raspy8766** [reply]
@rarkins Correct, our branch protections allow for any PR to merge with one approval which it has since we're using the `renovate-approve` bot. It's also worth noting all CI checks are passing as well for the example above.

As far as the `mergeStateStatus` value being incorrect, I've grabbed the [same query used by renovate](https://github.com/renovatebot/renovate/blob/91db1b5c61deb059673a6d2ace84a458bec0da4e/lib/platform/github/index.ts#L550) (hard-coding the repo/org name etc.) and I get a merge status like the following in response (see the response above in the dropdown): `"mergeStateStatus": "CLEAN",`. 

Also scanned for where the renovate bot's `"canMergeReason": "mergeStateStatus = BLOCKED"` debug log is created, and it appears to fall within [this `else` block](https://github.com/renovatebot/renovate/blob/main/lib/platform/github/index.ts#L568)

Given all that, I don't have any further debug information as to why that status would be coming through as `BLOCKED` for renovate bot and `CLEAN` when I use my own PAT.

**Reply by rarkins** [reply]
You have seen the source so you can see that Renovate doesn't "return" `mergeStateStatus` - GitHub's API does. 

Are you sure you don't have this option checked in your `master` branch protection?

![image](https://user-images.githubusercontent.com/6311784/129781656-4545be3a-2e8f-4f28-b2d5-dcc9212642c7.png)

Because if you do, and you are in that list of users, then it probably explains why GitHub returns clean for you but blocked for Renovate.

**Reply by rarkins** [reply]
And if that's not the case, please screenshot your completely branch protection settings for master branch and add them here

**Reply by raspy8766** [reply]
Right, by "returns" I meant the Github API returns "BLOCKED" for the renovate bot somehow. We do have branch protections/restrictions enabled, but with or without renovate and renovate-approve apps added, it still isn't auto-merging:

![image](https://user-images.githubusercontent.com/8533429/129815440-e01e5d3d-5eb2-4ea4-8fd1-fcdc5a79d34c.png)
![image](https://user-images.githubusercontent.com/8533429/129815486-58c92e6d-d006-4a98-bbdd-63ab6c873616.png)


It may also be worth noting, of the 3 types of repo's Github cloud offers (`internal`, `private`, `public`) we have both `internal` and `private` and auto-merge is not working for either type. The repo in this example is a `private` repo in our org

To debug this further I can briefly remove branch protections to see if that will enable auto-merge, but I will need to re-enable that per company/team policy, so it wouldn't be a long-term solution for us.

**Reply by rarkins** [reply]
Have those committer settings remained the same since you raised this issue or did you add renovate recently to that? I'm asking because with that setting it looks like only the bots could push to master branch, *maybe* also administrators (I'm not sure if "Include administrators" blocks that).

Either way, GitHub is telling Renovate that it's not permitted to merge, and we cannot bypass that.

**Reply by raspy8766** [reply]
So overnight that PR merged successfully! Turns out that I hadn't left the renovate and renovate bot in the "Restrict who can push to matching branches" long enough (tested it a while back) to let the renovate bot and renovate go through the merge cycle (CI checks => renovate approve => renovate merge).

As far as this question:
> it looks like only the bots could push to master branch

This is the setting that allows members of the org to merge without whitelisting members: 
![image](https://user-images.githubusercontent.com/8533429/129923159-02b4658c-388f-42bb-96ff-055d5094ab86.png)

My mistake was thinking that since the Renovate apps are installed in the org, they could/should be included as well. Anyways, that's not the case and now the problem is solved. I really appreciate all the help with this, thank you very much for your time.
